components/workflow.rst

```diff
@@ -19,6 +19,46 @@ You can install the component in 2 different ways:
 * :doc:`Install it via Composer </components/using_components>` (``symfony/workflow`` on `Packagist`_);
 * Use the official Git repository (https://github.com/symfony/workflow).
 
-For more information, see the code in the Git Repository.
+.. include:: /components/require_autoload.rst.inc
+
+Creating a Workflow
+-------------------
+
+The workflow component gives you an object oriented way to define a process
+or a life cycle that your object goes through. Each step or stage in the
+process is called a *place*. You do also define *transitions* to that describes
+the action to get from one place to another.
+
+.. image:: /_images/components/workflow/states_transitions.png
+
+A set of places and transitions creates a **definition**. A workflow needs
+a ``Definition`` and a way to write the states to the objects, (i.e. an
+instance of a :class:`Symfony\\Component\\Workflow\\MarkingStore\\MarkingStoreInterface`.
+
+Consider the following example for a blog post. A post can have places:
+'draft', 'review', 'rejected', 'published'. You can define the workflow
+like this::
+
+    use Symfony\Component\Workflow\Definition;
+    use Symfony\Component\Workflow\Transition;
+    use Symfony\Component\Workflow\Workflow;
+    use Symfony\Component\Workflow\MarkingStore\ScalarMarkingStore;
+
+    $states = ['draft', 'review', 'rejected', 'published'];
+
+    // Define a transaction with a name, where to go from and where to go to
+    $transitions[] = new Transition('to_review', 'draft', 'review');
+    $transitions[] = new Transition('publish', 'review', 'published');
+    $transitions[] = new Transition('reject', 'review', 'rejected');
+
+    $definition = new Definition($states, $transitions);
+    $definition->setInitialPlace('draft');
+
+    $marking = new ScalarMarkingStore('currentState');
+    $workflow = new Workflow($definition, $marking);
+
+The ``Workflow`` can now help you to decide what actions that are allowed
+on a blog post depending on what *place* it is in. This will keep your domain
```

Comment:
> BlogPost entity shouldn't be modified every time the rules of publishing evolve, for instance.

It's perfectly fine to change your domain model every time business logic of `BlogPost` changes. It's primary goal of the domain model to reflect business logic.

Otherwise your domain model is not a model, but bunch of properties with setters and getters.
http://www.martinfowler.com/bliki/AnemicDomainModel.html

I'd be more happy if this Workflow would trigger `$blog->sendOnReview()` for correspond transition(s) instead of hacking model internals. It would make more sense. However, it looks like the component is not designed for it.